modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -299,29 +300,32 @@ function social_group_request_preprocess_activity(array &$variables) {
   $activity = $variables['elements']['#activity'];
 
   if (!$activity->get('field_activity_message')->isEmpty()) {
+    $message_storage = \Drupal::entityTypeManager()->getStorage('message');
     /** @var Drupal\message\Entity\Message $message */
-    $message = $activity->field_activity_message->entity;
+    $message = $message_storage->load($activity->get('field_activity_message')->getValue()[0]['target_id']);
 
-    if (!$message->get('template')->isEmpty()) {
-      $message_template = $message->template->target_id;
+    if ($message instanceof Message) {
+      if (!$message->get('template')->isEmpty()) {
+        $message_template = $message->template->target_id;
 
-      if ($message_template === 'request_to_join_a_group') {
-        $activity_entity = $activity->getRelatedEntity();
+        if ($message_template === 'request_to_join_a_group') {
```

Comment:
It's better to replace it with a switch statement instead.

Reply:
@navneet0693 I think a switch will not work here, since our checks are not at the same level

Reply:
@DmytroRovovyi Switch is on basis of `$message_template`

Reply:
@navneet0693 I checked the switch. case statements are equivalent to the '==' operator, not the '===' operator, so the following code will not work.